Foxtrot/Driver/Rewriting/PostRewriteChecker.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Compiler;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Contracts.Foxtrot
+{
+    public sealed class PostRewriteChecker : Inspector
+    {
+        private Member currentMember;
+        private readonly Action<CompilerError> handleError;
+
+        public PostRewriteChecker(Action<CompilerError> handleError)
+        {
+            this.handleError = handleError;
+        }
+
+        public override void VisitProperty(Property property)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
+            {
+                this.currentMember = property;
+                exchanged = true;
+            }
+
+            base.VisitProperty(property);
+
+            if (exchanged)
+            {
+                this.currentMember = null;
+            }
+        }
+
+        public override void VisitEvent(Event evnt)
+        {
+            bool exchanged = false;
+            if (this.currentMember == null)
```

Comment:
Contracts may be defined in event and property accessors too. This code checks whether rewriter got it right in these places. Basically we only need VisitMemberBinding method to check correctness of the result. Three other methods (VisitEvent, VisitMethod, VisitProperty) provides a diagnostic information which memeber contains an invalid contract.